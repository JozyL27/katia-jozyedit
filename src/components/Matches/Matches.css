.matches__icons-container {
    display: flex;
    justify-content: space-around;
    align-items: center;
    padding: 1vmax 0;
    height: 8vmax;
    max-height: 7rem; 
    width: 80%;
    margin: auto;
}

.matches__back-icon {
    height: 4vmax;
    max-height: 5rem;
    margin: .5vmax 0;
}

.matches__header-container {
    display: flex;
    justify-content: center;
    flex-direction: column;
    align-items: center;
}

.matches__header {
    text-align: center;
    font-size: 2.8vmax;
    width: 20vmax;
    padding: .5vmax;
    color: black;
    border: none;
}

.headerBorder {
    border: .1rem solid black;
    width: 12%;
    margin-top: 2rem;
}

.matches__mail-icon {
    height: 5vh;
}

.matches__unmatch-icon {
    height: 5vh;
}

.matches__unmatch-icon:hover {
    cursor: pointer;
}

.matches__ul {
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    width: 80%;
    margin: auto;
}

.matches__li {
    cursor: pointer;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    margin: 1vmax 0;
    background: white;
    border-radius: .8vmax;
    width: 100%;
    box-shadow: .5rem 1rem .8rem #888888;
}

.matches__avatar {
    height: 15.5vmax;
    width: auto;
    max-width: 80%;
    border-radius: .8vmax;
    margin-bottom: 2vmax;
}

.matches__controls {
    width: 100%;
    display: flex;
    justify-content: space-evenly;
    align-items: center;
    padding: 1rem;
}

.matches__display-name {
    font-size: 3vmax;
    margin-bottom: 2vmax;
}

.matches__info {
    height: 100%;
    padding: 2vmax 0 0;
    width: 100%;
    transition: padding .1s, height .1s;
    transition-timing-function: ease-in-out;
}

.matches__info > p, .matches__gamer-tags > p {
    font-size: 2.3vmax;
    transition: font-size .3s;
    transition-delay: .2s;
    transition-timing-function: ease-in-out;
}

.matches__card-header {
    width: 100%;
    font-size: 2.5vmax;
    font-weight: bold;
    padding: .5vh 2vw;
    margin: .7vmax 0;
    transition: font-size .2s, background-color .2s, padding .2s, margin .2s;
    transition-timing-function: ease-in-out;
    text-align: center;
    color: black;
    border-bottom: .3rem solid lightgray;
    width: 60%;
    margin: 0 auto;
}

.matches__platforms {
    width: 100%;
    display: flex;
    justify-content: space-around;
    padding: 1vmax 0;
}

.matches__platforms > * {
    height: 6vmax;
    transition: height .5s;
    transition-delay: .1s;
}

.matches__genres, .matches__gamer-tags {
    font-size: 2.3vmax;
    text-align: center;
}

.matches__bio {
    font-size: 2.3vmax;
    text-align: center;
    padding: 0 1.5vmin;
    margin-bottom: 2vmax;
}

.hidden {
    padding: 0;
}

.matches__li > .hidden > div {
    padding: 0;
    transition-delay: 0.2s;
    border: none;
}

.matches__li > .hidden > h4 {
    font-size: 0;
    background-color: white;
    padding: 0;
    margin: 0;
    transition-delay: 0.1s;
    border: none;
}

.matches__li > .hidden > p {
    font-size: 0;
    padding: 0;
    margin: 0;
    transition-delay: 0s;
}

.hidden > .matches__platforms > * {
    height: 0;
    transition-delay: 0s;
}

.matchesStyleDiv {
    display: flex;
    flex-direction: column;
    align-items: center;
    width: 100%;
    border-radius: .8vmax;
    padding: 1rem;
    box-shadow: .5rem 1rem .8rem #888888;
    transition-duration: 0.8s;
}

.noMatchesPara {
    text-align: center;
    margin-top: 18rem;
    font-weight: bold;
    font-size: 1.6rem;
}

@media screen and (min-width: 600px) {
    .matches__icons-container, .matches__ul {
        max-width: 70%;
    }
}

@media screen and (min-width: 900px) {
    .matches__icons-container, .matches__ul {
        max-width: 55%;
    }

    .matchesStyleDiv {
        height: 250px;
    }

    .matchesStyleDiv:hover {
        box-shadow: .5rem 1rem .8rem lightskyblue;
    }

    .matches__info {
        width: 80%;
    }

    .matches__li {
        width: 70%;
    }

    .matches__header {
        font-size: 2rem;
    }

    .matches__back-icon {
        height: 3rem;
    }

    .headerBorder {
        width: 10%;
        margin-top: 0;
    }

    .matches__avatar {
        height: 100px;
        margin-bottom: 20px;
    }

    .matches__display-name {
        font-size: 25px;
        margin-bottom: 20px;
    }

    .matches__card-header.hidden1 {
        font-size: 22px;
    }

    .matches__card-header {
        font-size: 22px;
    }

    .matches__platforms > * {
        height: 4vmax;
    }

    .matches__info > p {
        font-size: 20px;
        margin-top: 10px;
        margin-bottom: 10px;
    }
}