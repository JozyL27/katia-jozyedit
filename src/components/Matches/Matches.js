import React from 'react'
import './Matches.css'
import UserContext from '../../Contexts/UserContext'
import { Link } from 'react-router-dom'
import nintendoLogo from '../../images/nintendo_logo.png'
import playstationLogo from '../../images/playstation_color_png.png'
import PC_Logo from '../../images/PC_Keyboard_Mouse_Icon.png'
import xboxLogo from '../../images/xbox_logo_png.png'
import mail_icon from '../../images/mail.svg'
import unmatch_icon from '../../images/x-circle.svg';
import MatchesService from '../../services/matches-service'
import backArrow from '../../images/left-arrow-svgrepo-com.svg'

import ConfirmDialog from '../Confirm/ConfirmDialog'
import customConfirm from '../../Utils/CustomConfirm'

export default class Matches extends React.Component {
    state = {
        users: [],
        error: null
    }

    static contextType = UserContext;

    componentDidMount(){
        MatchesService.getMatchedUsers(this.context.user_id)
            .then(res => this.setState({ users: res}))
            .catch(error => this.setState({error: error.message}))
    }

    toggleExpanded = (event) => {
        event.currentTarget.getElementsByClassName('matches__info')[0].classList.toggle('hidden')
    }

    handleMatchDelete = (e, id) => {
        e.stopPropagation();
        customConfirm(ConfirmDialog, 'Remove Match').then(answer => {
            if(answer) {
                MatchesService.removeMatch(this.context.user_id, id)
                    .then(() => {
                    let { users } = this.state;
                    users = users.filter(user => {
                        return user.user_id !== id;
                    });
                this.setState({ users });
            });
            }
        })
    }

    render() {
        let users = this.state.users || []
        const gamerTagPlatforms = ['Xbox', 'PSN', 'Nintendo', 'Steam', 'Discord', 'Other']

        return (
            <>
            <div className='matches__icons-container'>
                <Link to='/swipe' className='matches__Link'>
                    <img className='matches__back-icon' src={backArrow} alt='swipe-back-icon' />
                </Link>
                {/* **Uncomment when adding messages icon when messages component is added** <div className='matches__messages'></div> */}
            </div>
            <header className='matches__header-container'>
                <div className='headerBorder'></div>
                <h3 className='matches__header'>Matches</h3>
            </header>
            {this.state.users[0] === 'none' ? <p className='noMatchesPara'>You have no matches.</p> : 
            <ul className='matches__ul'>
                {users.map((user, index) => <li key={index} className='matches__li' onClick={this.toggleExpanded} >
                <div className='matchesStyleDiv'>
                    <h4 className='matches__display-name'>{user.display_name}</h4>
                    <img src={user.avatar} alt='avatar' className='matches__avatar'></img>
                    <div className='matches__controls'>
                        <Link to={`/chat/${user.user_id}`} className='matches__Link'>
                            <img className='matches__mail-icon' src={mail_icon} alt='chat-icon' />
                        </Link>
                        <img className='matches__unmatch-icon' src={unmatch_icon} alt='unmatch-icon' onClick={(e) => this.handleMatchDelete(e, user.user_id)} />
                    </div>
                </div>
                <section className='matches__info hidden'>

                    <h4 className='matches__card-header hidden1'>Platforms</h4>
                    <div className='matches__platforms'>
                        {user.platforms.includes("Xbox") ? <img className='matches__xbox' src={xboxLogo} alt='Xbox logo' /> : null}
                        {user.platforms.includes("PlayStation") ? <img className='matches__playstation' src={playstationLogo} alt='Playstation logo' /> : null}
                        {user.platforms.includes("Nintendo") ? <img className='matches__nintendo' src={nintendoLogo} alt='Nintendo logo' /> : null}
                        {user.platforms.includes("PC") ? <img className='matches__PC' src={PC_Logo} alt='PC logo' /> : null}
                    </div>

                    <h4 className='matches__card-header'>LFM In</h4>
                    {this.context.generateLfmElements(user.lfm_in)}

                    <h4 className='matches__card-header'>Genres</h4>
                    <p className='matches__genres'>{this.context.generateGenreString(user.genres)}</p>

                    <h4 className='matches__card-header'>Gamer Tags</h4>
                    {gamerTagPlatforms.map((platform, index) => {
                        // If the user has a gamer tag for that platform display it
                        return user[platform.toLowerCase()] !== '' && <p className='matches__gamer-tags' key={index}>{platform + ': ' + user[platform.toLowerCase()]}</p>
                    })}
                    
                    <h4 className='matches__card-header'>Bio</h4>
                    <p className='matches__bio'>{user.bio}</p>
                </section>
                </li>)}
            </ul>}
            </>
        )
    }
}