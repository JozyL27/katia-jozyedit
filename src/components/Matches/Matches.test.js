import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router-dom'
import Matches from './Matches'
import { UserProvider } from '../../Contexts/UserContext'

describe('<Matches />', () => {
  // Smoke test
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(
      <MemoryRouter>
        <UserProvider>
          <Matches />
        </UserProvider>
      </MemoryRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
  // Snapshot test
  it('renders the UI as expected', () => {
    const tree = renderer
      .create(
        <MemoryRouter>
          <UserProvider>
            <Matches />
          </UserProvider>
        </MemoryRouter>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})