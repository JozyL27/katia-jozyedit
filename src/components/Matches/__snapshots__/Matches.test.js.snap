// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Matches /> renders the UI as expected 1`] = `
Array [
  <div
    className="matches__icons-container"
  >
    <a
      className="matches__Link"
      href="/swipe"
      onClick={[Function]}
    >
      <img
        alt="swipe-back-icon"
        className="matches__back-icon"
        src="left-arrow-svgrepo-com.svg"
      />
    </a>
  </div>,
  <header
    className="matches__header-container"
  >
    <div
      className="headerBorder"
    />
    <h3
      className="matches__header"
    >
      Matches
    </h3>
  </header>,
  <ul
    className="matches__ul"
  />,
]
`;
